Automattic/wp-calypso
# ThemesListFetcher: Fix componentWillReceiveProps; Drop `themes-last-query` state subtree

Also, remove `key` prop from theme showcase's single-site mode. This was used to force `theme-list-fetcher` to update on site change. However, this was only required because `theme-list-fetcher`'s `componentWillReceiveProps` method had some funky checks which are now fixed.

Turns out we can drop the entire `themes-last-query` state subtree and replace it by more common selectors.

To test:
* In the theme showcase, switch back and forth between different sites (including Jetpack ones). Verify that the correct themes are shown. Furthermore, verify that searching and filtering still works fine.

Required for #9218
